### ISSUE TITLE
Keyword completion cache not updated.

### ISSUE THREAD

**Issue by kootstra** [nokia/RED#37]
Presently I'm working on extending the [Robotframework Sikuli library](https://github.com/rainmanwy/robotframework-SikuliLibrary) and I'm able to extend the library and use those new keywords successfully in the test cases. 

However, it seems that whenever I update the library, and install it via PIP, the new keywords are not recognized by RED and they are also not present in the keyword completion functionality of RED. The only way to overcome this is by creating a new project, import the sources and then the keyword completion does work.

To me it seems that RED caches the keywords of a (previously) loaded library. For performance reasons this makes sense, but sadly when removing the library and then adding it back again it does not update the cache. This could be considered a bug, or perhaps a feature request. In any case if anyone knows where these files are located so that I can remove them when needed would be great.

**Comment by KrzysztofJozefowicz** [comment]
Hi
"Changing libs from within RED and outside and reload" is already covered in RED.

For reloading libs on change action (only for main module file,not dependencies) there is option "Automatically reload changed libraries" in red.xml under Referenced libraries tab. You can also force to Project clean from Project menu.

RED/Eclipse does not track changes (yet) of any libs dependencies so if you make a change in dependent files you can manually reload RED Python env by RMC on project name, from Robot Framework use Reset Robot Environment. For special cases use Project clean and remove libspecs from project folder from disk. 

I will add those hints to RED help in next release as it seems to be missing. 
If above is useful you can close the issue.

**Comment by kootstra** [comment]
Not entirely sure what dependend libs are, and if the main jar file in the SikulyLibrary projects fits that descrition. If that is the case, then it explains it somewhat.

In any case I've been able to find a workaround. The information is cached in the LibSpec file, stored in: 
```
 \\workspace\\<project name>\libspecs\\SikuliLibrary.libspec
```
By removing this file followed by a Project Clean from the Project menu, I needed to add the library reference again and the new LibSpec file was generated with the additional keywords.

In my view refreshing the LibSpec files is desireable, and as this is at workspace level, having this connected to the Project Clean action might be a good approach.

**Comment by KrzysztofJozefowicz** [comment]
What I mean about lib dependencies is that currently RED tracks changes only at parent lib file (for instance sikuli.py from robotframework-SikuliLibrary ) but does not track any inherited files from sikuli.py ( generally all imports).
Therefore check if you have enabled lib reload option from red.xml and if there are any changes in dependent files try Reset Python Env action.

**Comment by KrzysztofJozefowicz** [comment]
Added info about lib reloading under Recognizing external libraries and variable files in RED in Help.
This will be released in v 0.7.1
